src/status_im/contexts/wallet/tokens/events.cljs

```diff
@@ -0,0 +1,142 @@
+ (fn [{:keys [db]} [{:keys [data]}]]
+   (let [chain-ids                              (chain/chain-ids db)
+         tokens                                 (reduce (fn [{:keys [by-address by-symbol] :as data}
+                                                             {:keys [name source version tokens]}]
+                                                          (-> data
+                                                              (update :sources
+                                                                      conj
+                                                                      {:name         name
+                                                                       :source       source
+                                                                       :version      version
+                                                                       :tokens-count (count tokens)})
+                                                              (update :by-address
+                                                                      merge
+                                                                      (tokens-data/tokens-by-address
+                                                                       {:added-tokens by-address
+                                                                        :source-name  name
+                                                                        :tokens       tokens
+                                                                        :chain-ids    chain-ids}))
+                                                              (update :by-symbol
+                                                                      merge
+                                                                      (tokens-data/tokens-by-symbol
+                                                                       {:added-tokens by-symbol
+                                                                        :source-name  name
+                                                                        :tokens       tokens
+                                                                        :chain-ids    chain-ids}))))
+                                                        {:sources    []
+                                                         :by-address {}
+                                                         :by-symbol  {}}
+                                                        data)
+         {:keys [sources by-address by-symbol]} tokens
+         symbols                                (->> by-symbol
+                                                     keys
+                                                     (remove wallet-validation/eth-address?))]
+     {:fx [[:effects.wallet.tokens/fetch-market-values
+            {:symbols    symbols
+             :currency   constants/profile-default-currency
+             :on-success #(rf/dispatch [:wallet.tokens/store-market-values %])
```

Comment:
If you use `utils.re-frame/call-continuation`, events using the effect `:effects.wallet.tokens/fetch-market-values` will be able to pass an event vector instead of an anonymous function, which greatly helps writing a simple equality check in the event test.